tests/unit/plugins/test_support/docker_image_archive_stubbing.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Red Hat | Ansible
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os.path
+import tarfile
+import tempfile
+
+
+def write_imitation_archive(file_name, image_id, repo_tags):
+    """
+    Write a tar file meeting these requirements:
+
+    * Has a file manifest.json
+    * manifest.json contains a one-element array
+    * The element has a Config property with "[image_id].json" as the value name
+
+    Args:
+        file_name: real file to write
+        image_id: fake sha256 hash (without the sha256: prefix)
+        repo_tags: list of fake image:tag's
+
+    """
+
+    manifest = [
+        {
+            'Config': '%s.json' % image_id,
+            'RepoTags': repo_tags
+        }
+    ]
+
+    write_imitation_archive_with_manifest(file_name, manifest)
+
+
+def write_imitation_archive_with_manifest(file_name, manifest):
+    manifest_file_name = tempfile.mkstemp(suffix='json')[1]
+
+    with open(manifest_file_name, 'w') as f:
+        f.write(json.dumps(manifest))
+
+    try:
+        tf = tarfile.open(file_name, 'w')
+        try:
+            ti = tarfile.TarInfo('manifest.json')
+            ti.size = os.path.getsize(manifest_file_name)
+
+            with open(manifest_file_name, 'rb') as f:
+                tf.addfile(ti, f)
+
+        finally:
+            # In Python 2.6, this does not have __exit__
+            tf.close()
+
+    finally:
+        os.remove(manifest_file_name)
+
+
+def write_irrelevant_tar(file_name):
+    inner_file_name = tempfile.mkstemp(suffix='txt')[1]
+
+    with open(inner_file_name, 'w') as f:
```

Comment:
Same here.

Reply:
Another great catch, made same two changes (your suggestion + moving inner file create/delete inside of tar file try..finally)